internal/fs/inode/dir_test.go

```diff
@@ -1450,3 +1452,40 @@ func (t *DirTest) LocalFileEntriesWithUnlinkedLocalChildFiles() {
 	AssertEq(1, len(entries))
 	AssertEq(entries[0].Name, "1_localChildInode")
 }
+
+func (t *DirTest) Test_ShouldInvalidateKernelDirCache_FirstCall() {
+	// By setting this, we can eliminate the case when time is initialized with
+	// first epoch value (1970).
+	ttl := util.MaxTimeDuration
+
+	shouldInvalidate := t.in.ShouldInvalidateKernelDirCache(ttl)
+
+	AssertEq(true, shouldInvalidate)
+}
+
+func (t *DirTest) Test_ShouldInvalidateKernelDirCache_True() {
+	_, err := t.readAllEntries()
+	AssertEq(nil, err)
+	ttl := time.Millisecond * 10
```

Comment:
nit: Technically this is not right that you're testing with ttl in milliseconds, which cannot even be set with out interface. Is it possible to set this in seconds at least and then speed up time using a simulated-clock?